test/tests/selftest/pbs_test_revert_site_hooks.py

```diff
@@ -0,0 +1,60 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.selftest import *
+
+
+class TestRevertSiteHook(TestSelf):
+    """
+    Contains test to delete hooks in Setup() except cgroups hook
+    """
+
+    def test_verify_delete_hooks_in_setup(self):
+        """
+        Test that creates any other site other than cgroups hook and
+        check if cgroups hook exists at the end of the test case
+        """
+        rc = self.server.manager(MGR_CMD_CREATE, HOOK, None, "exechook")
+        self.assertEqual(rc, 0)
+        TestSelf.setUp(self)
+        hooks = [h['id'] for h in self.server.status(HOOK)]
+        if len(hooks) != 1:
+            self.fail("Default cgroups hook doesn't exist")
+        else:
+            for h in hooks:
+                if h not in ['pbs_cgroups']:
+                    self.fail("Default cgroups hook doesn't exist")
```

Comment:
you have already confirmed that there is only one item in hooks, so the for loop is not required.
However, there is no guarantee that we would always have only cgroups enabled by default, if we enable another hook the test for len(hooks) to be 1 will fail. So, I suggest that you just check for existence of pbs_cgroups in "hooks" list.